src/dbnode/persist/fs/mmap_util.go

```diff
@@ -31,65 +31,67 @@ func validateAndMmap(
 	fdWithDigest digest.FdWithDigestReader,
 	expectedDigest uint32,
 	forceMmapMemory bool,
-) ([]byte, error) {
+	reporterOptions mmap.ReporterOptions,
```

Comment:
Same here, might be better to just propagate the `mmap.Reporter` then set the `ReporterOptions` here ourselves?

That way it's pretty clear and in control of setting up the `Context` itself, etc. Also not clear if it needs to add tags if needs to check if `Context` already has tags and to take a copy and modify, etc. Just seems cleaner to construct `ReporterOptions` and `Context` with the given reporter right before use.